spec/ics-2-consensus-requirements/README.md

```diff
@@ -0,0 +1,76 @@
+
+2. Headers have at least one direct child
+There is at least one direct child for all blocks, meaning that the lightclient logic can proceed the blocks one by one even in the worst case. If it not satisfied then there can be a point where the lightclient stops and cannot proceed, which halts IBC connection unexpectedly. This also can be violated when the blocks are restarted out of the consensus, for example in Tendermint, (1/2 < validators < 2/3) forked out the chain. This also will be covered in **byzantine recovery strategies**.
+
+3. If a block verifies another block then it is a descendant of the block
+Lightclient should not verify packets which are not in its chain. If the verifier returns true for a block that is not a descendant, it simply means that there is an error in the lightclient logic.
+
+4. Header can have a state only if it is a valid transition from the parent's(see the next paragraph)
+Header cannot have a state which is not a transition from its parent. This requires that the chain satisfies the IBC protocol and runs it honestly. In case where the validators become malicious and send invalid messages, the application logic on the destination chain should prevent further interchain infection.
+ 
+These requirements allow channels to work safely without concerning about double spending attack. 
+If a block is submitted to the chain(and optionally passed some challenge period for fraud proof), then it is assured that the packet is finalized so the application logic can process it.
+
+### Technical Specification
+
+In a real world, `Header`s do not have to contain the full state data; it can contain an aggregated data set identifier, for example a merkle root, where the actual data latter given can be proven that it is a member of that set. If we don't consider about the data availability problem, this partial blocks, which is be called `Header`s, still satisfies `Header`.
+
+Following functions exist over `Header`s.
+
+* `height : Header -> int`
+Returns the height of the block.
+
+* `verify : Header -> Header -> Maybe<LightClientProof> -> Maybe<error>`
+Verifies the latter block is a descendant of the former block, as defined in **lightclient specification**. Can take an additional lightclient proof argument.
+
+* `state : Header -> [MerkleProof] -> Error | KVStore`
+Returns a `KVStore` with the state inside the header. Can take Merkle proofs for the key-value pairs proving that the pairs are part of the header state.
+
+### Example Implementation
+
+An example blockchain `B` is run by a single operator. If a block is signed by the operator, then it is valid. `B` contains `KVStore`, which is a associative list with type of `[([byte], [byte])]`.  
+
+### Other Implementations
+
+* Cosmos-SDK: [](https://github.com/cosmos/cosmos-sdk/x/ibc)  
```

Comment:
Please update link to https://github.com/cosmos/cosmos-sdk/tree/develop/x/ibc or other desired branch (Provided link gives 404 and should have some text as well

Reply:
I think we can leave this out of the spec until it's actually implemented